hipcheck/src/plugin/manager.rs

```diff
@@ -3,13 +3,107 @@
 use crate::{
 	hc_error,
 	plugin::{try_get_bin_for_entrypoint, HcPluginClient, Plugin, PluginContext},
+	policy::policy_file::PolicyPluginName,
 	Result,
 };
 use futures::future::join_all;
-use hipcheck_common::proto::plugin_service_client::PluginServiceClient;
+use hipcheck_common::{proto::plugin_service_client::PluginServiceClient, types::LogLevel};
+use log::{debug, error, info, trace, warn};
 use rand::Rng;
-use std::{ffi::OsString, ops::Range, path::Path, process::Command};
-use tokio::time::{sleep_until, Duration, Instant};
+use serde::Deserialize;
+use std::{collections::HashMap, env, ffi::OsString, ops::Range, path::Path, sync::LazyLock};
+use tokio::process::Command;
+use tokio::{
+	io::{AsyncBufReadExt, BufReader},
+	time::{sleep_until, Duration, Instant},
+};
+
+static PLUGIN_LOG_SETTINGS: LazyLock<PluginLogLevels> =
+	LazyLock::new(|| PluginLogLevels::new_from_env().expect("Invalid log level var set"));
+
+/// structs for deserializing plugin logs
+#[derive(Debug, Deserialize)]
+struct LogEvent {
+	level: LogLevel,
+	fields: LogFields,
+	target: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct LogFields {
+	message: String,
+}
+
+pub struct PluginLogLevels {
+	plugin_wide_level: LogLevel,
+	plugin_level_map: HashMap<String, LogLevel>,
+}
+impl PluginLogLevels {
+	/// Parses HC_LOG env var to fetch log level for each plugin as Struct
+	pub fn new_from_env() -> Result<Self> {
+		// default level to error when var unset
+		let mut plugin_wide_level = LogLevel::Error;
+		let mut plugin_level_map: HashMap<String, LogLevel> = HashMap::new();
+
+		let Ok(rust_log) = env::var("HC_LOG") else {
+			return Ok(PluginLogLevels {
+				plugin_wide_level,
+				plugin_level_map,
+			});
+		};
+
+		let settings: Vec<&str> = rust_log.split(',').collect();
+		let mut plugin_wide_set = false;
+
+		for setting in settings {
+			let parts: Vec<&str> = setting.split('=').collect();
+
+			match (parts.first(), parts.get(1)) {
+				(Some(module), Some(level)) => {
+					let log_level: LogLevel =
+						LogLevel::level_from_str(level).expect("Invalid log level var");
+
+					if *module == "plugin" {
+						// e.g. HC_LOG="plugin=info"
+						// plugin wide log level is set, overriding global level
+						plugin_wide_level = log_level;
+						plugin_wide_set = true;
+					} else {
+						// e.g. HC_LOG="plugin::typo=info"
+						// plugin specific log level is found
+						plugin_level_map.insert(module.to_string(), log_level);
+					}
+				}
+				(Some(level), _) => {
+					// e.g. HC_LOG="info"
+					let log_level: LogLevel =
+						LogLevel::level_from_str(level).expect("Invalid log level var");
```

Comment:
(2/2) Since this is parsing user-typed data, there's a good chance this can fail in normal use.
So instead of panicking with `expect()`, this should return `Err()`.